05-UI-ReactJS/source/movieInput.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import {connect} from "react-redux";
+import {Link} from "react-router";
+import {addMovie, editMovie, initializeState} from "./redux/action";
+import {handleMovies} from "./redux/reducer";
+import Movie from "./movie";
+import store from "./store";
+
+class MovieInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    let movies = JSON.parse(this.props.params.movieID)
```

Comment:
What happens if the prop is not defined? There is a better way to set defaultProps. Check [here](https://babeljs.io/blog/2015/06/07/react-on-es6-plus)